### PR TITLE
BOAC-1608/BOAC-1609, Cohort view > proper results when user clicks 'Apply'; create cohort

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1608
https://jira.ets.berkeley.edu/jira/browse/BOAC-1609

* Include `student_count` when `create_cohort`
* `/api/cohort/create` accepts an array of filters, as seen in UI
* click 'Apply' and only the students section of the page gets a spinner

### CODE REVIEW

File: boac/api/cohort_controller.py [hunk 1]
```diff
@@ -107,35 +109,75 @@ def get_cohort(cohort_id):
...
+            'groupCodes',
+            'lastNameRange',
+            'levels',
+            'majors',
+            'underrepresented',
+            'unitRanges',
+        ]:
+            filter_criteria[key] = get_param(params, key)
+        for key in ['inIntensiveCohort', 'isInactiveAsc', 'isInactiveCoe']:
+            filter_criteria[key] = to_bool(params.get(key))
```
Comment (johncrossman) [inline boac/api/cohort_controller.py]: The `else` block above will be removed when Vue cutover is done. Going forward, the front-end only deals with a list of 'filters' and is not asked to deal with `filter_criteria` dict as stored in db.

File: boac/models/cohort_filter.py [hunk 2]
```diff
@@ -65,15 +65,11 @@ def __repr__(self):
...
     @classmethod
-    def create(cls, uid, name, **kwargs):
-        at_least_one_is_defined = False
-        filter_criteria = {}
-        for k, v in kwargs.items():
-            at_least_one_is_defined = at_least_one_is_defined or (len(v) if isinstance(v, list) else v is not None)
-            filter_criteria[util.camelize(k)] = v
-        if not at_least_one_is_defined:
+    def create(cls, uid, name, filter_criteria, student_count=None):
+        if all(not value for value in filter_criteria.values()):
```
Comment (paulkerschen) [inline boac/models/cohort_filter.py]: @johncrossman Will this logic get tripped up if searching on criteria with value `False`?

Comment (johncrossman) [reply boac/models/cohort_filter.py]: Yes, good catch. I'll amend

**Comment by paulkerschen** [comment]
Thanks for the array-tidying on the filters, @johncrossman.

**Pull request merged**